dxp/resource_serving.md

```diff
@@ -0,0 +1,161 @@
+# Resource Serving in DXP
+
+This document explains how different resources (`.js`, `.css`, etc.) are handled by DXP to be served to the browser.
+
+## Common Serving Architecture
+
+When it comes to serving anything from DXP there are two important files from which to start the investigation. Both are inside [portal-web/docroot/WEB-INF](https://github.com/liferay/liferay-portal/tree/master/portal-web/docroot/WEB-INF):
+
+1. [web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/web.xml): this is the standard `web.xml` file of DXP webapp. It defines the portlets and filters that should be loaded by the application server (by default Tomcat).
+2. [liferay-web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/liferay-web.xml): this is a clone of `web.xml` that is loaded by DXP once `com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter` is loaded.
+
+The default DXP's startup sequence goes like this:
+
+1. `Tomcat` is started.
+2. `Tomcat` bootstraps the `ROOT` application.
+3. The `ROOT` application initializes `com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter` and maps it to every URL requested.
+4. The `InvokerFilter` processes `liferay-web.xml` and behaves like a standard application server.
+
+This means that we only have to care about `liferay-web.xml` unless we need to find anything so low-level that we need to look at `web.xml`. It also means, that any request to DXP passes through `InvokerFilter` so it is a good place to put a breakpoint if we want to investigate any request routing.
```

Comment:
Super interesting. I had no idea it worked like this.

Reply:
Not so interesting (IMO), but yes :sweat_smile: